ci improvements

Fishrock123 opened the pull request:
In particular, also run ci on the `"hyper-client"` backend, and don't build docs for deps.

goto-bus-stop commented on .github/workflows/ci.yaml:
```diff
@@ -71,17 +71,16 @@ jobs:
-        override: true
+
+    - name: nightly setup
+      run: rustup toolchain install nightly
+      run: rustup default nightly
```
duplicate `run` key doesn't look right?

Fishrock123 replied:
No this is right, we could also do `cargo +nightly <command>` for the commands below.

goto-bus-stop replied:
sorry I sholuld've been clearer, the command looks right, but the duplicate `run` keys are not valid YAML

Fishrock123 replied:
Oh! Huh, I wonder why this worked at all then.

goto-bus-stop replied:
i dont think it did 😝 https://github.com/http-rs/surf/actions/runs/346519413

Fishrock123 replied:
It's because it ran no jobs, i.e. failed entirely.

goto-bus-stop commented:
LGTM once that ^ fixed 👍

Fishrock123 commented:
Updated, improved even a bit more. Finally merging this.

I have reverted the change to no longer use nightly for `fmt, clippy, docs` as it turns out that actions-rs/toolchain does indeed do downgrades if components are not found, or at least, it says it does: https://github.com/actions-rs/toolchain#components
